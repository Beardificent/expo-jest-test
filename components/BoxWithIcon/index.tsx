import { Box, HStack, Icon, Text } from "native-base";
import { FC } from "react";
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'

interface BoxWithIconProps {
  iconName: string;
  text: string;
}

const BoxWithIcon: FC<BoxWithIconProps> = ({ iconName, text }) => {
  return (
    <Box w={"fit-content"} borderWidth={1} p={2} data-testid="box-icon">
      <HStack alignItems={'center'}>
        <Icon
          color={"coolGray.700"}
          size={"2xl"}
          as={FontAwesome5}
          name={iconName}
        />
        <Text children={text} />
      </HStack>
    </Box>
  );
};

export default BoxWithIcon;
