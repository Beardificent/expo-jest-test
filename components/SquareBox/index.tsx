import { FC } from "react";
import { Box, View } from "native-base";
import React from "react";
import { INativebaseConfig } from "native-base"; // Update the import path for INativebaseConfig
import { NativeBaseProvider } from "native-base"; // Update the import path for NativeBaseConfigProvider

interface SquareBoxProps {
  children: any;
}
const SquareBox: FC<SquareBoxProps> = ({ children }) => {

  return (
      <Box w={'fit-content'} h={100} flexDir={'row'} backgroundColor={'blue.300'} borderWidth={1} p={2}>
        {children}
      </Box>
  );
};

export default SquareBox;
