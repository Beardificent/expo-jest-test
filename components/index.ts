import SquareBox from "./SquareBox";
import BoxWithIcon from "./BoxWithIcon";

export {SquareBox, BoxWithIcon}