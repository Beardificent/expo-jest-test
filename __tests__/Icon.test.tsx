import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { NativeBaseProvider } from "native-base";
import { BoxWithIcon } from "../components";

declare global {
  namespace JSX {
    interface IntrinsicElements {
      "mock-icon": typeof BoxWithIcon;
    }
  }
}

jest.mock("../components/BoxWithIcon/index.tsx", () => {
  const BoxWithIcon = (props: any) => (
    <mock-icon data-testid="mock-box-with-icon" {...props} />
  );
  return BoxWithIcon; // Return the component directly instead of an object
});

describe("BoxWithIcon", () => {
  it("renders BoxWithIcon with NativeBaseProvider + mocked component", () => {
    const inset = {
      frame: { x: 0, y: 0, width: 0, height: 0 },
      insets: { top: 0, left: 0, right: 0, bottom: 0 },
    };
    const expectedComponent = renderer.create(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <BoxWithIcon iconName="user" text="Hello BoxWithIcon" />
      </NativeBaseProvider>
    );
    expect(expectedComponent).toBeTruthy()
    expect(expectedComponent).toMatchSnapshot();
  });

  it("renders the component correctly without NativeBaseProvider + mocked content + correct property values", () => {
    const iconName = "user";
    const text = "Hello BoxWithIcon";

    const expectedComponent = renderer
      .create(<BoxWithIcon iconName={iconName} text={text} />)
      .toJSON() as ReactTestRendererJSON;

    expect(expectedComponent.props.iconName).toBe("user");
    expect(expectedComponent.props.text).toBe("Hello BoxWithIcon");
    expect(expectedComponent).toMatchSnapshot();
  });
});
