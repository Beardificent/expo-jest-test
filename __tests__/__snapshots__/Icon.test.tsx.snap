// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxWithIcon renders BoxWithIcon with NativeBaseProvider + mocked component 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <mock-icon
    data-testid="mock-box-with-icon"
    iconName="user"
    text="Hello BoxWithIcon"
  />
</RNCSafeAreaProvider>
`;

exports[`BoxWithIcon renders the component correctly without NativeBaseProvider + mocked content + correct property values 1`] = `
<mock-icon
  data-testid="mock-box-with-icon"
  iconName="user"
  text="Hello BoxWithIcon"
/>
`;
